### DISCUSSION TITLE
Open to the idea of having first class type support for certain standard library types such as std::array?

### DISCUSSION THREAD

**Discussion by 12345swordy** [hsutter/cppfront#1115] [Q&A]
I can not help to see the most obvious low hanging fruit is to implement the ability for us to write:
```
numbers: int[]= ( 1, 2 );
```
which is equivalent of this:
```
numbers: std::array<int,2> = ( 1, 2 );
```
That however requires first class type support for std::array in this instance. Thoughts?

**Comment by DyXel** [comment]
> That however requires first class type support for std::array in this instance. Thoughts?

Not a strange thing, first class support for standard types is something Cpp2 already embraces 🙂. For example: `new<T>` defaulting to what is essentially `std::make_unique<T>`.
